src/Features/Core/Portable/FeaturesResources.resx

```diff
@@ -171,6 +171,9 @@
   <data name="The_selection_contains_an_error_or_unknown_type" xml:space="preserve">
     <value>The selection contains an error or unknown type.</value>
   </data>
+  <data name="Cannot_extract_a_method_with_an_is_pattern_declaration" xml:space="preserve">
+    <value>Cannot extract a method with an is-pattern declaration.</value>
```

Comment:
Ah, i may be misunderstanding this.  Can you clarify which of the behaviors the user will have:

1. Extract method fails and does not allow them to do anything (giving the above message)
2. Extract method produces an error for the user, but still allows them to extract the method

If the latter, then i'm ok with this.  If the former, i think this is too restrictive.